addons/account/models/account_account.py

```diff
@@ -277,6 +297,76 @@ def _check_used_as_journal_default_debit_credit_account(self):
                 journal_ids=journals.ids
             ))
 
+    def _find_account_by_closest_parent_code(self, codes):
+        ''' Finds the account with the closest matching code
+            eg. code 123456 will return 123400 as it is a closer parent than 123756
+            :param codes: list(str) of codes to search for
+            :return: dict of codes with matching account values
+                {
+                    code1: {
+                        original_code: code1,
+                        match_len: int,
+                        account_id: matched account id,
+                        account_code: matched account code,
+                        user_type_id: matched account type
+                    }, ...
+                }
+        '''
+        if not codes:
+            return {}
+
+        place_holder_list = []
+        place_holder_vals = {}
+        for i, code in enumerate(codes):
+            place_holder_vals.update({f"code_{i}": code})
+            for j in range(len(code)):
+                place_holder_list.append(f"(%(code_{i})s, %(code_search_term_{i}_{j})s, %(code_match_len_{i}_{j})s)")
+                place_holder_vals.update({
+                    f"code_search_term_{i}_{j}": f"{code[:j + 1]}%%",
```

Comment:
You still do *not* need that double `%`, the only string interpolation you have is the f-string, then the resulting string is passed via `cr.execute()`'s kwargs.

The result is the same with 1 or 2 SQL wildcards, but it makes that awful piece of code even more confusing.